Internal: Restrict users from creating new Theme Builder parts when license is inactive [ED-7728]

mark-elementor opened the pull request.

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.5.0' into feature/container

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.5.0' into feature/container

StyleShit committed:
New: Added Container element [ED-2609] (#16105)

* New: WIP - Container element [ED-2609] (#14839)

* Internal: Release Container to dev-edition [ED-4274] (#16009)

* Internal: Container changes [ED-4863] (#16287)

* Hide floating bar [ED-4864]

* Add pointer to presets [ED-4869]

* Remove default nested container padding in preset [ED-4900]

* Don't nest Containers when dragging from panel [ED-4898]

* Improve width & height controls [ED-4865] [ED-4866]

* Use % for width by default [ED-4951]

* Change `height` to `min-height` [ED-4952]

* Fix: Print background overly CSS ":before" properties only when needed [ED-4965]

* Fix: Widget Editing handle is misaligned when Container is the first one in the page [ED-4953]

* Fix: Overflow issue when container is inside other container [ED-4964]

* Internal: Container changes [ED-5014] (#16429)

* Internal: Container changes [ED-2609] (#16655)

* Merge branch 'release/3.5.0' into dev/container (#16843)

* Tweak: Improve Container Drag & Drop [ED-5379] (#16772)

* Fix: Can't drag elements into Container [ED-5469] (#16860)

* Internal: Container fixes [ED-2609] (#16873)

* Internal: Container fixes [ED-2609] (#16878)

* Internal: Docs - Container element [ED-3809] (#15400)

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.5.0' into feature/container

matipojo committed:
Merge branch 'release/3.5.0' into feature/container

StyleShit committed:
Fix: D&D issues with right and left positioned elements [ED-5494] (#16959)

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.5.0' into feature/container

StyleShit committed:
Internal: Container Fixes [ED-5484] (#16964)

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.6.0' into feature/container

StyleShit committed:
Internal: Container fixes [ED-5605] (#17024)

iNewLegend committed:
Internal: Container - Fix `createContainerFromPreset` [ED-5713] (#17063)

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.6.0' into feature/container

StyleShit committed:
Tweak: Added "Convert to Container" control to each legacy section, inner section and page [ED-5488] (#17515)

iNewLegend committed:
Internal: Nested infra avoid defer for `getContainer()` [ED-6420] (#17662)

matipojo committed:
Merge branch 'release/3.6.0' into feature/container

iNewLegend committed:
Internal: Nested Infra new container `getParentAncestry` [ED-6415] (#17658)

iNewLegend committed:
Internal: Nested Infra new import to a specific container. [ED-6414] (#17657)

StyleShit committed:
Internal: Container Fixes [ED-5706] (#17459)

matipojo committed:
Merge remote-tracking branch 'elementor/develop' into feature/container

matipojo committed:
Internal: Use CommandHistory as external

The JS API shouldn't be used directly, in order to allow paths/names changes.

arielk committed:
Internal: Reorder flex-box settings [ED-6432]

arielk committed:
Tweak: Remove default Padding from the Container [ED-6366] 

* Fix: Center top-level container when Optimized DOM Output feature is enabled (#17415)

arielk committed:
Tweak: Convert "direction" control to icon based one [ED-5487]

arielk committed:
Tweak: Change Converter texts [ED-6434]

arielk committed:
Merge remote-tracking branch 'elementor/feature/container' into feature/container

arielk committed:
Internal: Add rearrange code

StyleShit committed:
Internal: Fix Container jest [ED-6485] (#17769)

StyleShit committed:
Internal: Container fixes [ED-6291] (#17790)

matipojo committed:
Merge branch 'release/3.6.0' into feature/container

iNewLegend committed:
Fix: tests (#17822)

matipojo committed:
Merge branch 'release/3.6.0' into feature/nested-infra

iNewLegend committed:
Internal: Nested Infra - CommandHistory `useHistory` option [ED-6553] (#17851)

StyleShit committed:
Internal: Container fixes [ED-6498] (#17895)

iNewLegend committed:
Internal: Nested infra avoid defer for getContainer() [ED-6420] (#17849)

iNewLegend committed:
Internal: Nested Infra - Allow creating widget(s) with custom `_title` [ED-6557] (#17857)

iNewLegend committed:
Internal: Nested Infra new ElementBase. [ED-6412] (#17654)

iNewLegend committed:
Tweak: Nested Infra - Allow dependencies for experiments [ED-6421] (#17663)

iNewLegend committed:
Internal: Container - Prepare feature/container to infra-nested [ED-6589] (#17950)

iNewLegend committed:
Internal: Nested Infra new partial widget render. [ED-6413] (#17655)

iNewLegend committed:
Internal: Nested Infra - Empty state - nested capabilities [ED-6555] (#17853)

iNewLegend committed:
Internal: Nested Infra - Frontend nested capabilities [ED-6554] (#17852)

iNewLegend committed:
Internal: Nested Infra - Fixes [ED-6599] (#17963)

iNewLegend committed:
Internal: Container - Prepare `feature/container` to `infra-nested`. (#17981)

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.7.0' into feature/nested-infra

matipojo committed:
Merge remote-tracking branch 'elementor/feature/container' into dev/nested-elements

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.6.0' into feature/nested-infra

matipojo committed:
Merge branch 'feature/nested-infra' into dev/nested-elements

iNewLegend committed:
Internal: Remove `stopPropagation` on edit widget [ED-6591] (#18040)

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.6.0' into feature/nested-infra

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.7.0' into feature/nested-infra

matipojo committed:
Merge branch 'release/3.7.0' into feature/nested-infra

iNewLegend committed:
Internal: Nested Infra - Add container to elementsManager [ED-6756] (#18093)

iNewLegend committed:
Internal: Nested Infra - Fix Experiments manager module dependency [ED-6757] (#18094)

iNewLegend committed:
Internal: Nested Infra - Fix Playwright [ED-6769] (#18128)

iNewLegend committed:
Tweak: Nested Infra - Select repeater item command [ED-6682] (#18039)

matipojo committed:
Merge remote-tracking branch 'elementor/feature/nested-infra' into dev/nested-elements

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.6.0' into dev/nested-elements

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.7.0' into dev/nested-elements

iNewLegend committed:
Merge remote-tracking branch '0-elementor/release/3.7.0' into feature/nested-infra

# Conflicts:
#	assets/dev/js/editor/components/settings/page/manager.js
#	assets/dev/js/editor/document/command-bases/command-history-base.js

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.7.0' into feature/nested-infra

iNewLegend committed:
Internal: Nested Elements - Infra fixes & tests [ED-6591] (#18297)

matipojo committed:
Revert "Internal: Nested Elements - Infra fixes & tests [ED-6591] (#18297)" (#18320)

This reverts commit 3c9addf02038fa5eff93e9d88ad5e7191c802396.

iNewLegend committed:
Fix: Nested Infra - Display Conditions window is blank [ED-6874] (#18301)

matipojo committed:
Merge branch 'release/3.7.0' into feature/nested-infra

iNewLegend committed:
Internal: Nested Elements - Infra fixes & tests [ED-6591] (#18373)

matipojo committed:
Merge remote-tracking branch 'elementor/feature/nested-infra' into dev/nested-elements

iNewLegend committed:
Merge remote-tracking branch '0-elementor/release/3.7.0' into dev/nested-elements

iNewLegend committed:
Remove: Duplicated method.

matipojo committed:
Merge remote-tracking branch 'iNewLegend/merge/3-7-0-to-nested-elements' into dev/nested-elements

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.7.0' into dev/nested-elements

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.7.0' into dev/nested-elements

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.7.0' into dev/nested-elements

iNewLegend committed:
New: Nested elements experiment [ED-2256] (#18030)

iNewLegend committed:
Fix playwright

iNewLegend committed:
Fix: Nested Elements - Playwright [ED-2256] (#18778)

matipojo committed:
Merge remote-tracking branch 'elementor/develop' into dev/nested-elements

matipojo committed:
Merge remote-tracking branch 'elementor/dev/nested-elements' into dev/nested-elements

iNewLegend committed:
Fix: Nested Tabs - Duplicate a Tab [ED-7328] (#18813)

TzviRabinovitch committed:
fixed deprecation message

iNewLegend committed:
Fix: Nested Tabs - Moving building block container cause tab to be empty. (#18830)

TzviRabinovitch committed:
Revert "New: Nested Tabs - Tab icons[ED-7329]"

This reverts commit c240a14ab2b4ab9291548ef7b988236279dff6a0.

TzviRabinovitch committed:
New: Nested Tabs - Tab icons[ED-7329]

iNewLegend committed:
Tweak: Tabs V2 - Changed experiment dependency [ED-7367] (#18863)

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.8.0' into dev/nested-elements

matipojo committed:
Revert "Internal: Nested Elements - Temporary disable the module"

This reverts commit 614a83467d6399e8efa828e76d299d7b9bec2ec8.

matipojo committed:
Internal: Nested Elements - Temporary disable the module

matipojo committed:
Merge branch 'release/3.8.0' into dev/nested-elements

TzviRabinovitch committed:
New: Nested Tabs - Tab icons [ED-7329] (#18866)

matipojo committed:
Internal: PHP Debug

matipojo committed:
Internal: PHP Debug #2

matipojo committed:
Internal: PHP Debug #3

matipojo committed:
Revert "Internal: PHP Debug #4"

This reverts commit e55d63ccfaee2192eae6e33bae1f29fecc1980a9.

matipojo committed:
Internal: PHP Debug #4

matipojo committed:
Revert Debug

iNewLegend committed:
Internal: Hide 'nested-elements' experiment [ED-2256] (#18928)

iNewLegend committed:
New: Nested Tabs - Use the old tabs if the experiment is Off [ED-7325] [ED-7326] (#18800)

Co-authored-by: Mati Horowitz <21468434+matipojo@users.noreply.github.com>

hein-obox committed:
New: Nested Tabs - Drag a new tabs widget onto the editor [ED-7327] (#18920)

udidol committed:
Internal: Show Nested Elements experiment in admin [ED-7562] (#19033)

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.8.0' into dev/nested-elements

TzviRabinovitch committed:
Merge remote-tracking branch 'elementor/release/3.7.0' into dev/nested-elements

hein-obox committed:
New: Nested Tabs - Style title section [ED-7333] (#19023)

hein-obox committed:
New: Nested Tabs - Style Tab [ED-7332] (#19003)

Co-authored-by: Hein van Vlastuin <vv@abanganimedia.co.za>
Co-authored-by: udidol <36724484+udidol@users.noreply.github.com>

Nevoss committed:
Fix: Nested Elements - Lint issues (#19052)

Nevoss committed:
Fix: Nested Elements - Sync repeater with containers [ED-7545] (#19015)

Nevoss committed:
Fix: Nested Element - Lock tab items [ED-7546] (#19046)

hein-obox committed:
New: Nested Tabs - Style the box section [ED-7335] (#19057)

Co-authored-by: Hein van Vlastuin <vv@abanganimedia.co.za>
Co-authored-by: udidol <36724484+udidol@users.noreply.github.com>

hein-obox committed:
New: Nested Tabs - Style icon section [ED-7334] (#19043)

Co-authored-by: Hein van Vlastuin <vv@abanganimedia.co.za>
Co-authored-by: udidol <36724484+udidol@users.noreply.github.com>

mark-elementor committed:
Internal: Restrict users from being able to create new Theme Parts via Theme Builder when license is inactive - Core [ED-7728]

mark-elementor committed:
Remove account service

mark-elementor committed:
Dependencies

mark-elementor committed:
Updates

mark-elementor committed:
Fix

mark-elementor committed:
Merge branch 'release/3.9.0' into ED-7728-app-license-service

mark-elementor committed:
Base service

mark-elementor committed:
Legacy theme builder fix

davseve commented on includes/template-library/sources/local.php:
```diff
@@ -1341,16 +1342,24 @@ public function print_blank_state_template( $current_type_label, $href, $descrip
 	}
 
+	private function is_license_valid() {
+		return ( new License_Service )->is_valid();
```
Why do you create a new instance of the License_Service? 
Haven't created a services class that registers all the services classes?

mark-elementor committed:
Small changes

mark-elementor committed:
Merge branch 'release/3.9.0' into ED-7728-app-license-service

mark-elementor committed:
Account service linked

ManorHazaz commented on includes/template-library/sources/local.php:
```diff
@@ -1718,6 +1732,7 @@ private function remove_elementor_cpt_from_sitemap( array $post_types ) {
 	public function __construct() {
 		parent::__construct();
 
+		$this->app_services = new AppServices();
```
Are you creating a new instance of the services for each usage?
 Don't you think it will be better to create the service once the general app.php is loaded and consume it from there?
 EDITED
 I saw you created the instances both in the JS and in PHP, but you didn't use it here, why?

ManorHazaz commented on app/services/license/license-service.js:
```diff
@@ -0,0 +1,17 @@
+		return 'app/services/license';
+	}
+
+	isActive() {
```
I'm assuming that you are planning to overwrite this from the pro, what will happen if the user won't update the pro plugin?

ManorHazaz commented on app/services/base-service.php:
```diff
@@ -0,0 +1,5 @@
+<?php
+namespace Elementor\App\Services;
+
+abstract class Base_Service implements Service_Interface {
```
Agreed, one of them is also redundant in my opinion.
Not sure which is the best, again you need to test it with the pro implementation. What will happen in the Pro if you want to implement the interface or extend the base which will not be available since to user only updated the pro?

ManorHazaz reviewed (commented):
This is not a full CR.
Those are just a few comments and concerns about the implementation of the services.

mark-elementor committed:
Merge branch 'release/3.9.0' into ED-7728-app-license-service

StyleShit committed:
Fix: RTL text is printed backwards in Text path widget (#17309) [ED-7661] (#20230)

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

mark-elementor committed:
Remove interface

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

mark-elementor committed:
Merge branch 'release/3.9.0' into ED-7728-app-license-service

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

mark-elementor committed:
Remove linebreak

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

mark-elementor committed:
Services hook

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

TzviRabinovitch committed:
Merge remote-tracking branch 'elementor/dev/nested-elements' into release/3.10.0

louiswol94 committed:
Fix: Duplicated loop takes me auto to the 1st loop (core) [ED-8648] (#20310)

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

ManorHazaz commented on app/services/services.php:
```diff
@@ -0,0 +1,39 @@
+	 * @param $name
+	 * @return mixed|null
+	 */
+	public function get_service( $name ) {
```
Maybe put the public functions first? just for readability.

hein-obox committed:
Tweak: Changed Typography control position ... [ED-8327] (#20348)

StyleShit committed:
Fix: Unnecessary "non-existing control" errors are thrown (#20027) [ED-8590] (#20358)

The `update_control()` method has been called with nullish values
without validation.

arielk committed:
Internal: Change notification banner updates text (#20381)

Co-authored-by: Yakir Sitbon <kingyes1@gmail.com>

yotamselementor committed:
Internal: Upgrade to Node 18 [ED-8708] (#20377)

Nevoss committed:
Fix: Nested Elements - Cannot drag widget inside the tab to the outside [ED-7665] (#20386)

TzviRabinovitch committed:
Merge remote-tracking branch 'elementor/release/3.9.0' into release/3.10.0

mark-elementor committed:
CR Fixes

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

hein-obox committed:
Tweak: Remove the Folder icon when using a nested tab [ed-8314] (#20353)

hein-obox committed:
Tweak: Adjusted the initial design of tabs [ED-8536] (#20378)

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

hein-obox committed:
Tweak: Updated the content in nested tabs [ED-8546] (#20366)

hein-obox committed:
Fix: Title Alignment is not working in Nested Tabs widget [ed-8317] (#20403)

hein-obox committed:
Tweak: Removed various elements from Nested Tabs widget [ed-8639] (#20352)

TzviRabinovitch committed:
Merge branch 'release/3.9.0' into release/3.10.0

ManorHazaz commented on app/services/base-service.php:
```diff
@@ -0,0 +1,7 @@
+namespace Elementor\App\Services;
+
+abstract class Base_Service {
+	abstract function get_name() : string;
+	abstract function register();
```
Add phpdoc

mark-elementor replied:
Added.

ManorHazaz commented on app/app.php:
```diff
@@ -16,9 +17,12 @@
 }
 
 class App extends BaseApp {
+	public static $instance;
```
Why?

mark-elementor replied:
Explained

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

hein-obox committed:
New: Nested Tabs - Change prefix for selectors [ed-7540] (#20451)

matipojo committed:
Internal: Preparing for PHPCS rules updates [ED-8756] (#20392)

KingYes committed:
Tweak: Merged several experiments to core [ED-8769] (#20462)

Co-authored-by: Ariel Klikstein <ariel@elementor.com>

StyleShit committed:
Internal: Preparing for PHPCS rules updates (2) [ED-8757] (#20396)

matipojo committed:
Merge remote-tracking branch 'elementor/release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

matipojo committed:
Internal: Bring back phpcs.xml deprecated file, for BC (#20464)

hein-obox committed:
New: Nested Tabs - Change prefix for selectors (v2) [ED-7540] (#20466)

matipojo committed:
Fix: Some Ajax request doesn't work as expected after ED-8756 [ED-8800] (#20469)

StyleShit committed:
Fix: Post type is overridden on autosave for library documents (#19941) [ED-8591] (#20421)

The bug was caused due to this part:

https://github.com/elementor/elementor/blob/e8d4d693c3d59ee77aa69d97f397e0aeb342995e/core/documents-manager.php#L150-L154
Each library document has overridden the CPT for all of the others
document types, so the last one was the default.


Closes #19941

Co-authored-by: Mati Horowitz <21468434+matipojo@users.noreply.github.com>

rami-elementor committed:
Fix: Translatable button in the experiments page [ED-8805] (#20482)

hein-obox committed:
New: Nested Tabs - Change prefix for selectors v3 [ed-7540] (#20485)

mark-elementor commented on app/services/services.php:
```diff
@@ -0,0 +1,39 @@
+	}
+
+	private function register_services() {
+		$this->services[ 'license' ] = ( new License_Service() )->register();
```
Done

mark-elementor committed:
CR fixes

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

rami-elementor committed:
Tweak: Don't use i18n function when throwing dependency exceptions [ED-8808] (#20488)

davseve committed:
Fix: Icon change position not centered & SVG icon is not visible in FE [ED-8316] (#20463)

Co-authored-by: udidol <36724484+udidol@users.noreply.github.com>

hein-obox committed:
Fix: Title Alignment is not working in Nested Tabs widget v2 [ed-8317] (#20491)

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

TzviRabinovitch committed:
New: Nested Tabs - Nested Elements experiment [ED-7533] (#20399)

hein-obox committed:
New: Nested Tabs - Responsive section [ED-7331] (#20417)

Co-authored-by: Hein van Vlastuin <vv@abanganimedia.co.za>
Co-authored-by: udidol <36724484+udidol@users.noreply.github.com>

elementorbot committed:
Merge branch 'release/3.9.0' into release/3.10.0

rami-elementor committed:
Tweak: Don't use i18n function when throwing exceptions [ED-8808] (#20490)

davseve committed:
New: Nested tabs - adjust spacing controls in responsive mode [ED-7558] (#20489)

mark-elementor committed:
Fix

mark-elementor committed:
Merge branch 'release/3.10.0' into ED-7728-app-license-service
